mindsdb/integrations/handlers/s3_handler/s3_handler.py

```diff
@@ -29,14 +28,25 @@ class ListFilesTable(APIResource):
     def list(self,
              targets: List[str] = None,
              conditions: List[FilterCondition] = None,
+             limit: int = None,
              *args, **kwargs) -> pd.DataFrame:
 
-        tables = self.handler._get_tables()
+        buckets = None
+        for condition in conditions:
+            if condition.column == 'bucket':
+                if condition.op == FilterOperator.IN:
+                    buckets = condition.value
+                elif condition.op == FilterOperator.EQUAL:
+                    buckets = [condition.value]
+                condition.applied = True
+
         data = []
-        for path in tables:
+        for obj in self.handler.get_objects(limit=limit, buckets=buckets):
```

Comment:
Is the `files` table expected to return all of the files in the bucket or only those that are supported for querying?

Reply:
the `files` returns all files for which you can get content (not dependent on file type. they all visible in `files` table)